### ISSUE TITLE
Improve settings tooltip accessibility

### ISSUE THREAD

**Issue by thatrobotdev** [manga-download/hakuneko#5557]
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Describe the bug**
In the settings dialog, the information mark that displays the description for each menu item (added in #2298, feature #2240 cc @Sheepux) is [inaccessible to assistive technologies, touch only devices, keyboard only users, people experiencing fine motor control impairment, people with cognitive concerns, etc.](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#accessibility_concerns)

**To Reproduce**
1. Click the hamburger (triple bar) button in the top left of the HakuNeko interface.
2. Next to any menu item in the "Settings" section, click on the information mark button.
3. Nothing appears on click or will appear if you wait, hovering. (bug)
4. Hover over tooltip to show item description in tooltip box (current implementation)

**Expected behavior**
At first I thought the intended behavior of only showing the item description on hover was a UX bug, as [I thought the tooltip would act like a toggletip](https://inclusive-components.design/tooltips-toggletips/) (toggletips display form descriptions on click instead of on hover). It took me a good minute to realize that I needed to hover instead of click on the tooltip to see the item description, and clicking in my browser's implementation on the tooltip would actually hide the item description indefinitely!

There are two behaviors I can think of that would be better than the current implementation,
1. [Don't use a tooltip/toggletip for the settings menu, and just provide item descriptions inline.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role#best_practices) I know that this would use up a lot of space though, so this might not be the best option in this case.
2. [Create a toggletip](https://inclusive-components.design/tooltips-toggletips/) that would show the item description on click in a cute little HakuNeko theme supporting styled box.

**Screenshots**
<figure>
    <img width="511" alt="Screenshot of HakuNeko's settings page with an arrow pointing to a tooltip" src="https://user-images.githubusercontent.com/18013689/224458402-a2e4e121-c1eb-4c1f-adcd-9f0f442d1c0b.png">
    <figcaption>This is the tooltip box in the Settings menu.</figcaption>
</figure>

<figure>
    <img width="511" alt="Screenshot of HakuNeko's settings page with an arrow pointing to the current implementation of the tooltip description box" src="https://user-images.githubusercontent.com/18013689/224459044-a6118199-ddec-4fc3-9eda-6ba956cc9ec7.png">
    <figcaption>The current implementation of the tooltip box.</figcaption>
</figure>

**Context (please complete the following information):**
 - OS: macOS 13.0.1
 - HakuNeko Version 6.17

**Additional context**
_Relevant lines of code_
https://github.com/manga-download/hakuneko/blob/99f9e91d75eb6d3d8c146c31e51378e2b4eac4e5/src/web/lib/hakuneko/frontend%40classic-light/menu.html#L197-L206

Tooltip code for light theme.

https://github.com/manga-download/hakuneko/blob/99f9e91d75eb6d3d8c146c31e51378e2b4eac4e5/src/web/lib/hakuneko/frontend%40classic-dark/menu.html#L197-L206

Tooltip code for dark theme.

---
I would love to help work on this! :)

**Comment by MikeZeDev** [comment]
That is definitely something to be considered. That said i have no idea about how haku interface works, and the people who knows are busy working on haruneko. If its not solved in Haku at least it should be considered in Haru.

**Comment by Sheepux** [comment]
It's "solved" (please test) in haruneko using IBM tooltips (it's a toggle)
see interface preview here https://haruneko.pages.dev/ (or start it from the github https://github.com/manga-download/haruneko )

Yes i have a11y (accessibility) & ARIA in mind for haru but as i have to learn/think from scratch there is currently too much for me on my little time (i'm not a dev :P)

Settings display are managed like this:
https://github.com/manga-download/haruneko/blob/master/web/src/frontend/classic/components/settings/InputChoice.svelte

```html
<SettingItem
    labelText={$Locale[setting.Label]()}
    helperText={$Locale[setting.Description]()}
>
    <Select
        selected={value}
        on:update={(evt) => (setting.Value = evt.detail.toString())}
    >
        {#each setting.Options as option}
            <SelectItem value={option.key} text={$Locale[option.label]()} />
        {/each}
    </Select>
</SettingItem>
```

Setting item is here https://github.com/manga-download/haruneko/blob/master/web/src/frontend/classic/components/settings/SettingItem.svelte
and is using carbon's tooltip just before the slot (the input previously described) https://carbon-components-svelte.onrender.com/components/Tooltip
```html
<div class={`menu-left-panel-${type}-container`}>
    {#if type === 'item'}
        <Tooltip triggerText={labelText} align="start">
            <p>{helperText}</p>
        </Tooltip>
    {/if}

    <slot />
</div>
```
keyboard navigation is requested in https://github.com/manga-download/hakuneko/issues/2009 and is tracked in haruneko in https://github.com/manga-download/haruneko/issues/40

If you want to help you can join the herd of the "frontenders" (i'd be glad to have some help), i'm now using svelte for the frontend in haruneko's rewrite from scratch (ongoing for 2 years)

**Comment by thatrobotdev** [comment]
Thanks for your reply, harukenko looks great! I'll definitely be following the other frontend work there!

I noticed there are some situations where tooltips display cut off.

<img width="1214" alt="Screenshot of Haruneko settings menu" src="https://user-images.githubusercontent.com/18013689/224580811-5925fee1-a585-4cd0-8606-881dc99dede8.png">

For example, the last tooltip in the "General" and "Style" panels both get cut-off by the bottom of the window, and creates new space for the tooltip that you need to scroll down to read the entire tooltip. Sometimes the tooltip will open instantaneously then close, but I couldn't figure out how it chooses between making more space and just closing it instantaneously.

<img width="255" alt="Screenshot of Haruneko quick settings panel" src="https://user-images.githubusercontent.com/18013689/224580912-6431504c-c7e3-470a-951d-9335e4405e6b.png">

In this quick settings panel, every tooltip is cut off by the end of the window.

It looks like as a limitation of the current implementation of the tooltip in the IBM library, we would need to manually change the direction of certain tooltips to avoid this (https://github.com/carbon-design-system/carbon-components-svelte/issues/623). I haven't used that much svelte before, so I'm not sure what the best way to implement an override on these components would be, but it could be a good workaround.

**Comment by Sheepux** [comment]
the last tooltips isn't cut but it has extra spacing at the end of the document (normal behavior), you can see the scrollbar.
I guess i should set directions on the right side in the global settings and on the left side in the right pannel settings
